lib/ansible/template/__init__.py

```diff
@@ -399,6 +399,11 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
         if fail_on_undefined is None:
             fail_on_undefined = self._fail_on_undefined_errors
 
+        if not fail_on_undefined:
+            display.warning(msg="Raising an error on undefined variables is set to False."
```

Comment:
this warning should not be default, iirc there are several paths that use fail_on_undefined and should not warn